tests/suites/test_suite_ssl.function

```diff
@@ -95,16 +95,16 @@ int mbedtls_test_buffer_put( mbedtls_test_buffer *buf,
 }
 
 /*
- * Gets \p output_len bytes from the \p output buffer into the ring buffer
- * \p buf.
+ * Gets \p output_len bytes from the ring buffer \p buf into the
+ * \p output buffer. The output buffer can be NULL, in this case a part of the
+ * ring buffer will be dropped, if the requested length is available.
  *
  * \p buf must have been initialized and set up by calling
  * `mbedtls_test_buffer_init()` and `mbedtls_test_buffer_setup()`.
  *
  * \retval  \p output_len, if the data is available.
  * \retval  0 <= value < \p output_len, if the data is not available.
- * \retval  -1, if \buf is NULL, it hasn't been set up or \p output_len is not
- *          zero and \p output is NULL
+ * \retval  -1, if \buf is NULL or it hasn't been set up.
```

Comment:
Ah, I misunderstood these changes yesterday. I thought this was changing how the message queue works and thought it should be done in the commit that added the message queue. However, these are changes to the already existing test buffer code. These changes indeed need their own commit to explain and justify why they are needed. Then, we can add the new socket mock tests in their own commit, given that the proper foundation will have been laid. Sorry for the back and forth.

Reply:
No problemo. Done!